internal/person/person_test.go

```diff
@@ -0,0 +1,245 @@
+		Height: 188,
+		Weight: 90,
+	}
+	tay := Person{
+		Name:   "Tay",
+		Age:    18,
+		Height: 190,
+		Weight: 87,
+	}
+	test := anna.IsAdult()
+	wantedResult := false
+	if test != wantedResult {
+
+		t.Errorf("anna.IsAdult() = %v; want %v", test, wantedResult)
+	}
+	test2 := luna.IsAdult()
+	wantedResult2 := true
+	if test2 != wantedResult2 {
+
+		t.Errorf("luna.IsAdult() = %v; want %v", test2, wantedResult2)
+	}
+	test3 := jan.IsAdult()
+	wantedResult3 := true
+	if test3 != wantedResult3 {
+
+		t.Errorf("jan.IsAdult() = %v; want %v", test3, wantedResult3)
+	}
+	test4 := tay.IsAdult()
+	wantedResult4 := true
+	if test4 != wantedResult4 {
+
+		t.Errorf("tay.IsAdult() = %v; want %v", test4, wantedResult4)
+	}
+}
+
+func TestAverage(t *testing.T) {
+	team := People{
+		{Name: "Anna",
+			Age:    17,
+			Height: 160,
+			Weight: 49,
+		},
+		{
+			Name:   "Luna",
+			Age:    30,
+			Height: 156,
+			Weight: 48,
+		},
+		{
+			Name:   "Jan",
+			Age:    29,
+			Height: 188,
+			Weight: 90,
+		},
+		{
+			Name:   "Tay",
+			Age:    33,
+			Height: 190,
+			Weight: 87,
+		},
+	}
+	test := team.Average()
+	wantedResult := Person{
+		Name:   "AveragePerson",
+		Age:    27,
+		Height: 173.5,
+		Weight: 68.5,
+	}
+	if test != wantedResult {
+
+		t.Errorf("team.Average() = %v; want %v", test, wantedResult)
+	}
+}
+
+func TestOldest(t *testing.T) {
+	team := People{
+		{Name: "Anna",
+			Age:    17,
+			Height: 160,
+			Weight: 49,
+		},
+		{
+			Name:   "Luna",
+			Age:    30,
+			Height: 156,
+			Weight: 48,
+		},
+		{
+			Name:   "Jan",
+			Age:    29,
+			Height: 188,
+			Weight: 90,
+		},
+		{
+			Name:   "Tay",
+			Age:    33,
+			Height: 190,
+			Weight: 87,
+		},
+	}
+	test := team.Oldest()
+	// "{Tay 33 190 87}"
+	wantedResult := team[3]
+	if test != wantedResult {
+
+		t.Errorf("team.Oldest() = %v; want %v", test, wantedResult)
+	}
+}
+
+func TestPeopleGreet(t *testing.T) {
+	team := People{
+		{Name: "Anna",
+			Age:    17,
+			Height: 160,
+			Weight: 49,
+		},
+		{
+			Name:   "Luna",
+			Age:    30,
+			Height: 156,
+			Weight: 48,
+		},
+		{
+			Name:   "Jan",
+			Age:    29,
+			Height: 188,
+			Weight: 90,
+		},
+		{
+			Name:   "Tay",
+			Age:    33,
+			Height: 190,
+			Weight: 87,
+		},
+	}
+	test := team.Greet()
+	namesSlice := []string{}
+	wantedResult := []string{}
+	for _, firstName := range team {
+		namesSlice = append(namesSlice, firstName.Name)
+	}
+
+	for _, value := range namesSlice {
+		for_, familyName := range namesSlice {
+			if value == familyName {
+				continue
+			}
+		}
+	}
+	// wantedResult := []string{"Hi Anna from Luna!",
+	// 	"Hi Anna from Jan!",
+	// 	"Hi Anna from Tay!",
+	// 	"Hi Luna from Anna!",
+	// 	"Hi Luna from Jan!",
+	// 	"Hi Luna from Tacdy!",
+	// 	"Hi Jan from Anna!",
+	// 	"Hi Jan from Luna!",
+	// 	"Hi Jan from Tay!",
+	// 	"Hi Tay from Anna!",
+	// 	"Hi Tay from Luna!",
+	// 	"Hi Tay from Jan!"}
+	// if test != wantedResult {
+
+		t.Errorf("team.Greet() = %v; want %v", test, wantedResult)
+	}
+}
```

Comment:
@nobe4 Unsure what to do for this one. I seem to not be able to just fill the slice manually, as it apparently can only be compared to nil.
Do I loop then?
Just format the the results as a string in array syntax?

Reply:
That's true you cannot compare slices directly, but you can try using https://pkg.go.dev/slices#Equal

e.g. https://go.dev/play/p/Jde00Lv27wv

Reply:
Attempt done.